web/modules/custom/adms_validator/src/Form/AdmsValidatorForm.php

```diff
@@ -0,0 +1,224 @@
+use EasyRdf\Sparql\Result;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form to validate the ADMS compliance of a RDF or TTL file.
+ */
+class AdmsValidatorForm extends FormBase {
+
+  /**
+   * The name of the graph used for validation.
+   *
+   * @var string
+   */
+  const VALIDATION_GRAPH = 'http://adms-validator/';
+
+  /**
+   * The path of the file that contains the validation rules.
+   *
+   * @var string
+   */
+  const SEMIC_VALIDATION_QUERY_PATH = "SEMICeu/adms-ap_validator/python-rule-generator/ADMS-AP Rules .txt";
+
+  /**
+   * The Sparql endpoint.
+   *
+   * @var \Drupal\rdf_entity\Database\Driver\sparql\Connection
+   */
+  protected $sparqlendpoint;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('sparql_endpoint')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\Connection $sparql_endpoint
+   *   The Sparql endpoint.
+   */
+  public function __construct(Connection $sparql_endpoint) {
+    $this->sparqlendpoint = $sparql_endpoint;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'adms_validator_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['adms_file'] = [
+      '#type' => 'file',
+      '#title' => $this->t('File'),
+      '#description' => $this->t('An RDF file you want to test for compliance.'),
+      '#upload_validators'  => [
+        'file_validate_extensions' => ['rdf ttl'],
+      ],
+    ];
+    $form['info'] = [
+      '#markup' => $this->t('This validator uses the <a href="https://github.com/SEMICeu/adms-ap_validator">SEMIC ADMS-AP ruleset</a>.'),
+    ];
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Upload'),
+      '#button_type' => 'primary',
+    ];
+    $info = $form_state->getBuildInfo();
+    if (!empty($info['validation_errors'])) {
+      // The form was submitted, and validation errors have been set.
+      $form['table'] = $this->buildErrorTable($info['validation_errors']);
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+    // Make sure to clear the table.
+    $form_state->addBuildInfo('validation_errors', []);
+    $form_state->setRebuild(TRUE);
+
+    $file = $this->uploadedFile();
+    if (!$file) {
+      $form_state->setError($form['adms_file'], 'Please upload a valid RDF file.');
+      return;
+    }
+    if (!$this->storeInGraph($file)) {
+      $form_state->setError($form['adms_file'], 'The provided file is not a valid RDF file.');
+      return;
+    }
+    // Delete the uploaded file from disk.
+    $file->delete();
+    $form_state->addBuildInfo('validation_errors', $this->getValidationErrors());
```

Comment:
We can use the `FormState::set()` to keep data in the storage or even a property in the form class, like the entity form does.